### ISSUE TITLE
Panel: Restore aria-expanded on the button

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#24036]
**Describe the bug**

In https://github.com/WordPress/gutenberg/pull/23327 the Panel component went through some refactoring. Amongst other things, the `aria-expanded` attribute was removed from the Panel button.

Now, all the panels in the editor don't give any clue to assistive technology users whether they're collapsed or expanded.

The `aria-expanded` attribute is of fundamental importance not only to communicate the Panel state but also to let users understand how they have to interact with it.

**To reproduce**
- test with the plugin version 8.5.1
- inspect the sidebar panels HTML in your browser's dev tools
- observe the panel buttons don't have an `aria-expanded` attribute any longer
- use a screen reader, navigate to the panel buttons and observe the collapsed / expanded state of the panels is not announced
